### ISSUE TITLE
Filename pattern Unicode?

### ISSUE THREAD

**Issue by zcyzcy88** [TheRealMarnes/doubleclick-image-downloader#15]
Filename pattern:
`%original% - %title%`

Test case:
https://www.google.com/search?q=%E5%95%8A%E5%93%88&newwindow=1&client=firefox-b-ab&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiUk8Xjv_zTAhULzmMKHb4nBToQ_AUICigB&biw=1480&bih=844#imgrc=49haW2vNcoChzM

result:
`201604210942233841718 - - - Google Search.png`

I can't find the Unicode character in result filename.

**Comment by TheRealMarnes** [comment]
unicode should just work in js, I'll try to find the cause...

**Comment by eight04** [comment]
https://github.com/TheRealMarnes/doubleclick-image-downloader/blob/master/src/index.js#L28
```js
const illegalCharsRegex = /[^\w\-\s\.,%;]+/ig;
```
Unicode doesn't inside the white list.

**Comment by TheRealMarnes** [comment]
I'm not sure how well allowing any and all unicode chars will play with e.g. windows... I'm sure linux and all would be fine, but I have little trust in windows handling it well. I don't want to end up allowing filesystem errors, undeletable files, etc to be spawned

**Comment by eight04** [comment]
Windows can handle unicode but there are some chars which are forbidden: https://stackoverflow.com/a/31976060

Also, trailing dots doesn't work well on Windows (e.g. `C:\foo.txt.`)

**Comment by TheRealMarnes** [comment]
Well yeah, but then there are all the thousands of unicode chars that are technically allowed. A lot of fonts don't even have them and won't render them. I imagine users downloading files with cyrillic, chinese, etc filenames and ending up with folders of files named ▯▯▯▯▯▯▯▯▯▯.xyz

**Comment by eight04** [comment]
It should be fine on Windows. If Firefox displays the fonts correctly, windows explorer would also work. They both use the fonts from the system.

**Comment by TheRealMarnes** [comment]
Yeah, it's all supposed to be alright, but I've honestly always considered unicode fonts to be one of those things that may work for the majority of people while an unlucky few get mercilessly screwed by it. I dunno, thin ice, unknown territory, too big and opaque to control, etc etc.

You know what, I'll add in unicode chars but provide a sort of "ascii safespace" switch in the prefs to disallow unicode for people finding themselves hindered by it.
